### PR TITLE
FEATURE: ONE-2347 omit exludeObjectWithTags option when includeObject…

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
…sWithTags is defined

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/397/ : SUCCESS in 2m 36s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/162/ : SUCCESS in 1m 21s

File: src/catalogue.js [hunk 1]
```diff
@@ -124,13 +124,19 @@ export function loadDateDataSets(projectId, options) {
         bucketItems = bucketItemsToExecConfig(bucketItems, { removeDateItems: true });
     }
 
+    const omittedOptions = ['filter', 'types', 'paging', 'dataSetIdentifier', 'returnAllDateDataSets', 'returnAllRelatedDateDataSets'];
+    // includeObjectsWithTags has higher priority than exludeObjectsWithTags, so when present omit exludeObjectsWithTags
```
Comment (pbenes) [inline src/catalogue.js]: maybe we could add a test here https://github.com/gooddata/gooddata-js/blob/master/test/catalogue_test.js to have the desired behaviour described by a test.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/398/ : SUCCESS in 2m 40s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/163/ : SUCCESS in 1m 44s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/399/ : FAILURE in 1m 05s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/164/ : FAILURE in 26s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/400/ : SUCCESS in 2m 02s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/165/ : SUCCESS in 31s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/401/ : SUCCESS in 1m 22s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/166/ : SUCCESS in 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/404/ : SUCCESS in 2m 07s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/169/ : SUCCESS in 1m 18s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/17/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/405/ : SUCCESS in 2m 59s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/170/ : SUCCESS in 3m 12s

**Pull request merged**